internal/pkg/agent/application/upgrade/step_download.go

```diff
@@ -54,6 +57,10 @@ func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI stri
 		return "", errors.New(err, "initiating fetcher")
 	}
 
+	if err := os.MkdirAll(paths.Downloads(), 0750); err != nil {
+		return "", errors.New(err, fmt.Sprintf("failed to create download directory at %s", paths.Downloads()))
+	}
```

Comment:
@michalpristas the downloader implementation looks like it does this internally: 

https://github.com/elastic/elastic-agent/blob/b8553c59a5a7c0b1944eee96562892fc748b0668/internal/pkg/agent/application/upgrade/artifact/download/http/downloader.go#L179-L183

What am I missing here?

Reply:
Looks like the change here may not have actually been necessary speaking with Michal.